wasi-sdk.cmake

```diff
@@ -1,7 +1,7 @@
 # Cmake toolchain description file for the Makefile
 
 # This is arbitrary, AFAIK, for now.
-cmake_minimum_required(VERSION 3.4.0)
```

Comment:
Is this file (along with the Platform file) now being used as part of the build?  Or maybe its being used to build the examples?

Reply:
Correct yeah, this was done since CMake was spewing a warning-per-compile (aka lots) that CMake itself is dropping support for 3.4.0 so I bumped this to 3.5.0 to stop CMake from warning.

Reply:
FYI, once https://github.com/WebAssembly/wasi-sdk/pull/430 is merged, we'll want to make the same change to wasi-sdk-p2.cmake